frontend/src/components/G6Graph.svelte

```diff
@@ -36,127 +36,256 @@
+      }); // the type of link is connected to the label of the edge here.
+    }
 
-        timeBarData.push({
-            date: ev.date,
-            value: "1",
-        });
-    };
+    edgesToBack(ev); // put all edges attached to the node behind the nodes (needed when using groupByTypes: false);
 
-    export const updateTimeBar = (timeBarEnabled: boolean) => {
-        graph.removePlugin(graph.get("plugins")[0]);
-        if (!timeBarEnabled) {
-            //TO-DO Reset the graph if wanted later
-        } else {
-            graph!.addPlugin(
-                new G6.TimeBar({
-                    className: "g6TimeBar",
-                    x: 0,
-                    y: 0,
-                    width: 500,
-                    height: 110,
-                    padding: 10,
-                    type: "trend",
-                    changeData: false,
-                    trend: {
-                        data: timeBarData,
-                        smooth: true,
-                    },
-                    tick: {
-                        tickLabelFormatter: (timeBarData: any) => {
-                            return "";
-                        },
-
-                        tickLineStyle: {
-                            fill: "#f28c18",
-                        },
-                    },
-                    slider: {
-                        backgroundStyle: {
-                            fill: "#131616",
-                        },
-                        foregroundStyle: {
-                            fill: "#ffffff",
-                        },
-                        handlerStyle: {
-                            style: {
-                                fill: "#f28c18",
-                                stroke: "#f28c18",
-                            },
-                        },
-                    },
-                    controllerCfg: {
-                        fill: "#131616",
-                        stroke: "#131616",
-                        timePointControllerText: " Point",
-                        timeRangeControllerText: " Point",
-                    },
-                })
-            );
-        }
-    };
+    timeBarData.push({
+      date: ev.date,
+      value: "1",
+    });
+  };
 
-    onMount(() => {
-        if (graph) {
-            graph.destroy();
-        }
+  /**
+   * Helper method that will rearrange the order of items on the z-index (edges behind the nodes)
+   * To avoid iterating through the whole graph and update, we use this helper method inside the push method.
+   * This allows to only manipulate the nodes newly pushed into the graph.
+   * @param event containing a node and its edges.
+   */
+  const edgesToBack = (event) => {
+    const node = graph.findById(event.id);
+    if (node instanceof Node) {
+      const edges = node.getEdges();
 
-        graph = new G6.Graph({
-            ...options,
-            container,
-        });
+      for(let i = 0; i < edges.length; i++){
```

Comment:
I mean a ranged for, I get why you were reluctant to use this but javascript has a modern range for:
```js
for (const edge of edges) {}
```

Reply:
Aaah! That makes more sense haha! Will do!

Reply:
This is how the code would look like in that case 

```
      for(const edge in edges){
        edges[edge].toBack();
      }
```

The problem with using `for ... in` is that the `const` is of type string and we need to use it as index value to make it work (which does not really add readability. 
But we can use `for ... of` as follow for more clarity? 

```
      for(const edge of edges){
        edge.toBack();
      }
```

Let me know which one you prefer =) (I would suggest the second one `for ... of`

Reply:
youre confusing me but just do the one that looks nice

Reply:
Haha, sorry, I'll go for the second one then :D